### PR TITLE
Deprecate nqp-lib

### PR DESCRIPTION
**Pull request by patrickbkr**
The option has been effect less for at least a year now. So remove all uses in Rakudo and add a deprecation notice.

This is for *after* the release.

### CODE REVIEW

**Comment by patrickbkr** [comment]
This was an accidental push of latest master to this branch. That's why it got autoclosed. I'm reopening.

File: src/Perl6/Compiler.nqp [hunk 1]
```diff
@@ -54,6 +54,10 @@ class Perl6::Compiler is HLL::Compiler {
             %options<doc> := 'Text';
         }
 
+        if nqp::existskey(%options, 'nqp-lib') {
+            note('Option `--nqp-lib` is deprecated, has no effect and will be removed in a future release.');
```
Comment (Altai-man) [inline src/Perl6/Compiler.nqp:58]: "Will be removed in 2021.06".

**Comment by patrickbkr** [comment]
@Altai-man I think it turned into a noop during my big relocatability refactor a year ago. But I'm not sure anymore. It might actually have been a noop already.

**Comment by Altai-man** [comment]
>I think it turned into a noop during my big relocatability refactor a year ago. But I'm not sure anymore. It might actually have been a noop already

No problems with squash-merge it once CI is green.

File: src/Perl6/Compiler.nqp (hunk 1 above)
Comment (patrickbkr) [reply src/Perl6/Compiler.nqp]: Resolved.

**Pull request merged**